channels/tee-channel-js/Tests/Unit/src/TeeChannelCore.Tests.ts

```diff
@@ -17,6 +17,48 @@ export class TeeChannelCoreTests extends AITestClass {
 
     public registerTests() {
 
+        this.testCase({
+            name: "config.channel dynamic config",
+            useFakeTimers: true,
+            test: () => {
+                const channelPlugin = new ChannelPlugin();
+                const teeChannel = new TeeChannel();
+                const appInsightsCore = new AppInsightsCore();
+                const channelPlugin1 = new ChannelPlugin();
+                channelPlugin1.priority = 1030;
+                const channelPlugin2 = new ChannelPlugin();
+                channelPlugin2.priority = 1031;
+
+                appInsightsCore.initialize(
+                    { instrumentationKey: "09465199-12AA-4124-817F-544738CC7C41", channels: [[channelPlugin, teeChannel],[channelPlugin1]]},[]
+                );
+                let coreChannels =  appInsightsCore.getChannels();
+                Assert.equal(2, coreChannels.length, "Total number of core channel queues");
+                const teeChannels = teeChannel.getTeeChannels();
+                Assert.equal(1, teeChannels.length, "Total number of tee channel queues");
+                Assert.equal(channelPlugin1, teeChannels[0][0], "Total number of tee channel queues");
+
+                
+                appInsightsCore.config.extensionConfig = appInsightsCore.config.extensionConfig? appInsightsCore.config.extensionConfig : {};
+                let extConfig = appInsightsCore.config.extensionConfig;
+                let teeChanneConfig = extConfig[teeChannel.identifier];
+                Assert.deepEqual(teeChanneConfig, {ignoreCoreChannels: false, teeChannels: null}, "default config is set");
+                appInsightsCore.config.extensionConfig[teeChannel.identifier] = {ignoreCoreChannels: false, teeChannels: [[channelPlugin2]]};
+                this.clock.tick(1);
+                coreChannels = appInsightsCore.getChannels();
+                Assert.equal(2, coreChannels.length, "Total number of core channel queues");
+                let channel = teeChannel.getTeeChannels();
```

Comment:
Oh, this is interesting...
Question: Looking at the code this looks like this sequence would include itself as it's own TeeChannel -- that would not end well. We should fix this so that a TeeChannel doesn't add itself to it's set of channels so that we don't end up with a recursive issue.